docs/styleguide.md

```diff
@@ -0,0 +1,198 @@
+Before you read this document, you should familiarize yourself with the upstream [yocto project styleguide](https://www.openembedded.org/wiki/Styleguide). meta-nilrt roughly follows the upstream guide, except where the rules in this document disagree.
+
+
+# General
+
+**Capitalize NI LinuxRT and NILRT**
+
+When referring to "NI LinuxRT" in recipes and documentation, or its abbreviated form: "NILRT", capitalize the N-I-L-R-and-T. The exception to this rule is when speaking about the "meta-nilrt" layer itself, or the "nilrt" git repository - where it is recommended you call it "nilrt.git".
+
+**Indentation**
+
+Use **tabs** for indentation and **spaces** for alignment in all places - *except* within python task functions, where spaces should be used for both.
```

Comment:
Why the deviation here?

Reply:
Because I believe strongly that soft tabs are degenerate, and also upstream isn't actually consistent about using soft tabs.

Reply:
It sounds like a personal opinion. Even if upstream doesn't enforce the soft tabs, if it is in their style guide, I think we should follow it. It'll be less of a hassle for developers who follow the style guide and wish to contribute to multiple layers.

Reply:
Like I said, upstream doesn't actually enforce this style and you can find many examples of both in upstream. Is this a blocker for you, because I feel strongly about it, but I don't want to go through the effort of selling it to upstream to change on the wiki?

Reply:
Sorry, I missed this response. I do feel strongly about enforcing this policy since I think it is not sustainable. I think we should only care about consistency within a file or write a linter to do the work for us. But since you've already made all of those changes based on what you felt, I'm not going to block the way.